microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/SubsystemDefinition.java

```diff
@@ -32,42 +32,77 @@
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.microprofile.opentracing.smallrye.TracerConfigurationConstants;
 
 public class SubsystemDefinition extends PersistentResourceDefinition {
+
     private static final String OPENTRACING_CAPABILITY_NAME = "org.wildfly.microprofile.opentracing";
+    public static final String DEFAULT_TRACER_CAPABILITY_NAME = "org.wildfly.microprofile.opentracing.default-tracer";
 
     private static final RuntimeCapability<Void> OPENTRACING_CAPABILITY = RuntimeCapability.Builder
             .of(OPENTRACING_CAPABILITY_NAME)
             .addRequirements(WELD_CAPABILITY_NAME)
             .build();
 
+    public static final RuntimeCapability<Void> DEFAULT_TRACER_CAPABILITY = RuntimeCapability.Builder
+            .of(DEFAULT_TRACER_CAPABILITY_NAME, true)
+            .build();
+
+    public static final RuntimeCapability<Void> TRACER_CAPABILITY = RuntimeCapability.Builder
+            .of(TRACER_CAPABILITY_NAME, true, OpentracingConfigurationService.class)
+            .build();
+
     static final String[] MODULES = {
-            "io.jaegertracing.jaeger",
-            "io.opentracing.contrib.opentracing-tracerresolver",
-            "io.opentracing.opentracing-api",
-            "io.opentracing.opentracing-util",
-            "org.eclipse.microprofile.opentracing",
-            "org.eclipse.microprofile.restclient"
+        "io.opentracing.contrib.opentracing-tracerresolver",
+        "io.opentracing.opentracing-api",
+        "io.opentracing.opentracing-util",
+        "org.eclipse.microprofile.opentracing",
+        "org.eclipse.microprofile.restclient"
     };
 
     static final String[] EXPORTED_MODULES = {
-            "io.smallrye.opentracing",
-            "org.wildfly.microprofile.opentracing-smallrye",
-            "io.opentracing.contrib.opentracing-interceptors",
-    };
+        "io.smallrye.opentracing",
+        "org.wildfly.microprofile.opentracing-smallrye",
+        "io.opentracing.contrib.opentracing-interceptors",};
+
+    public static final SimpleAttributeDefinition DEFAULT_TRACER = SimpleAttributeDefinitionBuilder
+            .create(TracerConfigurationConstants.DEFAULT_TRACER, ModelType.STRING, true)
+            .setCapabilityReference(TRACER_CAPABILITY_NAME)
+            .setRestartAllServices()
+            .build();
+
     protected SubsystemDefinition() {
-        super( new SimpleResourceDefinition.Parameters(SubsystemExtension.SUBSYSTEM_PATH, SubsystemExtension.getResourceDescriptionResolver(SubsystemExtension.SUBSYSTEM_NAME))
+        super(new SimpleResourceDefinition.Parameters(SubsystemExtension.SUBSYSTEM_PATH, SubsystemExtension.getResourceDescriptionResolver())
                 .setAddHandler(SubsystemAdd.INSTANCE)
-                .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE)
+                .setRemoveHandler(new SubsystemRemoveHandler())
                 .setCapabilities(OPENTRACING_CAPABILITY)
```

Comment:
Should you not also register DEFAULT_TRACER_CAPABILITY?

Reply:
No because this default is only set if the default attribute (which is nullable) is set